From 524d9bebf06439048072e5266fca3cb7ce2aacc1 Mon Sep 17 00:00:00 2001
From: Felix Yan <felixonmars@archlinux.org>
Date: Sun, 29 Aug 2021 20:11:28 +0800
Subject: [PATCH] Define ITT_ARCH_IA64 when undefiend

I am not sure why ITT_ARCH_IA64 was skipped here. This results in unsupported architecture (RISC-V, in my case) falls to the `ITT_ARCH==ITT_ARCH_IA64` branch below (because both of them are undefined) and results in a build error due to `__TBB_machine_fetchadd4` being undefined.
---
 src/tbb/tools_api/ittnotify_config.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/tbb/tools_api/ittnotify_config.h b/src/tbb/tools_api/ittnotify_config.h
index 8ecc23781..f904a8e9d 100644
--- a/src/tbb/tools_api/ittnotify_config.h
+++ b/src/tbb/tools_api/ittnotify_config.h
@@ -147,6 +147,10 @@
 #  define ITT_ARCH_IA32E 2
 #endif /* ITT_ARCH_IA32E */
 
+#ifndef ITT_ARCH_IA64
+#  define ITT_ARCH_IA64 3
+#endif /* ITT_ARCH_IA64 */
+
 #ifndef ITT_ARCH_ARM
 #  define ITT_ARCH_ARM  4
 #endif /* ITT_ARCH_ARM */
